stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -98,75 +92,114 @@ public void store(String name, String value) {
     return result;
   }
 
+  @SuppressLint("DefaultLocale")
   @ChromeDevtoolsMethod
   public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params) {
     final GetMatchedStylesForNodeRequest request = mObjectMapper.convertValue(
         params,
         GetMatchedStylesForNodeRequest.class);
 
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+    result.matchedCSSRules = new ArrayList<>();
+    result.inherited = Collections.emptyList();
+    result.pseudoElements = Collections.emptyList();
+
     mDocument.postAndWait(new Runnable() {
       @Override
       public void run() {
-        Object elementForNodeId = mDocument.getElementForNodeId(request.nodeId);
+        final Object elementForNodeId = mDocument.getElementForNodeId(request.nodeId);
 
         if (elementForNodeId == null) {
           LogUtil.w("Failed to get style of an element that does not exist, nodeid=" +
               request.nodeId);
           return;
         }
 
-        mDocument.getElementStyles(
-            elementForNodeId,
-            new StyleAccumulator() {
+        mDocument.getElementStyleRules(elementForNodeId, new StyleRuleAccumulator() {
+          @Override
+          public void store(String ruleName, boolean editable) {
+            final ArrayList<CSSProperty> properties = new ArrayList<>();
+
+            final RuleMatch match = new RuleMatch();
+            match.matchingSelectors = ListUtil.newImmutableList(0);
+
+            final Selector selector = new Selector();
+            selector.value = ruleName;
+
+            final CSSRule rule = new CSSRule();
+            rule.origin = Origin.REGULAR;
+            rule.selectorList = new SelectorList();
+            rule.selectorList.selectors = ListUtil.newImmutableList(selector);
+            rule.style = new CSSStyle();
+            rule.style.cssProperties = properties;
+            rule.style.shorthandEntries = Collections.emptyList();
+
+            if (editable) {
+              rule.style.styleSheetId = String.format("%d.%s", request.nodeId, selector.value);
```

Comment:
in order to save an int->Object boxing allocation, which will relieve some GC pressure, prefer this pattern:

`rule.style.styleSheetId = String.format("%s.%s", Integer.toString(request.nodeId), selector.value);`